packit/upstream.py

```diff
@@ -466,11 +467,49 @@ def fix_spec(self, archive: str, version: str, commit: str):
         :param version: version to set in the spec
         :param commit: commit to set in the changelog
         """
+        # we only care about the first number in the release
+        # so that we can re-run `packit srpm`
+        original_release_number = self.specfile.get_release_number().split(".", 1)[0]
+
+        current_time = datetime.datetime.now().strftime("%Y%m%d%H%M%S%f")
+
+        git_des_command = [
+            "git",
+            "describe",
+            "--tags",
+            "--long",
+            "--dirty",
+            "--match",
+            "*",
+        ]
+        try:
+            git_des_out = run_command(git_des_command, output=True).strip()
+        except PackitCommandFailedError as ex:
+            logger.info(f"Exception while describing the repository: {ex}")
+            # probably no tags in the git repo
+            git_desc_suffix = ""
+        else:
+            # git adds various info in the output separated by -
+            # so let's just drop version and reuse everything else
+            g_desc_raw = git_des_out.split("-", 1)[1]
```

Comment:
Will this work for all projects? Are tags always versions only? The reason for asking is b/c in git.centos.org, I see tags like `imports/c8/rpm-4.14.2-25.el8`, which also contain a version, but they will fail to be processed this way.

Reply:
that's git-describe specific, it uses dashes to separate different data - sadly one cannot format it

yeah, we may need to polish this for centos stream - or figure out how to operate with release and version - we will be in control of the repos, so we may use more sensible tags

Reply:
what I wrote is bullshit, I thought we're talking about something else - in your case, curse all the people who use dashes in tags